### DISCUSSION TITLE
Record LED

### DISCUSSION THREAD

**Discussion by NRKorsakov** [shanteacontrols/OpenDeck#309] [Q&A]
Hi! Is there a way to wire up something to make a lighted "record" button?

I know this can be done with Mackie HUI/MCU, which is not supported here, and the old Midi MMC protocol, which unfortunately sort of takes over your transport controls in a negative way.

What I wanted to build is a switch or foot-switch that would punch in & out of record in my DAW, but also light an LED above it when it is in-record mode.

Thanks!

**Comment by paradajz** [comment]
If your DAW can send back some of the MIDI messages supported by the LEDs then yes? Not really sure what the problem is here.

**Comment by NRKorsakov** [comment]
I have to mess with this more.

I had always assumed that the only way to get 2-way communication like this is with the Mackie HUI or MCU protocol. MMC does some of this also, but I don't like the way it "takes over" your DAW.

2-way communication with DAWs maybe has progressed further than I am assuming, but I don't see how DAWs send out that kind of information rather than just taking it in.

Thanks for the reply. I'll mess around with it some more!

**Reply by paradajz** [reply]
AFAIK Ableton can send out CC messages out when certain functions change. Don't know the details.

**Comment by NRKorsakov** [comment]
Hi thank you for the reply! I'm using Motu DP. There are some things in DP I have not explored, as well as Mackie MCU and Eucon support. The Mackie MCU protocol, which has been open-sourced or at least reverse-engineered, is actually pretty cool, despite being written eons ago. It actually must be somewhat open-source as there are quite a few products based on it out there that aren't manufactured by Mackie. I kind of wrote it off as inferior to Eucon, but when I tried one and also an Avid Artist, the Mackie protocol actually worked really well, and in some ways (at least for me) better if you are used to organizing your DAW for production.
